iOS/BugsnagTests/BugsnagKSCrashSysInfoParserTest.m

```diff
@@ -0,0 +1,36 @@
+//
+//  BugsnagKSCrashSysInfoParserTestTest.m
+//  Tests
+//
+//  Created by Jamie Lynch on 15/05/2018.
+//  Copyright © 2018 Bugsnag. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+
+#import "BugsnagKSCrashSysInfoParser.h"
+
+@interface BugsnagKSCrashSysInfoParserTest : XCTestCase
+@end
+
+@implementation BugsnagKSCrashSysInfoParserTest
+
+- (void)testEmptyDictSerialisation {
+    // ensures that an empty dictionary parameter returns a fallback dictionary populated with at least some information
+    NSDictionary *device = BSGParseDevice(@{});
+    XCTAssertNotNil(device);
+    XCTAssertNotNil(device[@"locale"]);
+    XCTAssertNotNil(device[@"freeDisk"]);
+    XCTAssertNotNil(device[@"simulator"]);
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+
+- (void)testNilDictSerialisation {
+    XCTAssertNotNil(BSGParseDevice(nil));
```

Comment:
This could return an empty dictionary and still pass. I think this should check the contents too.